python/cudf/utils/cudautils.py

```diff
@@ -610,6 +610,33 @@ def gpu_diff(in_col, out_col, N):
             out_col[i] = -1
 
 
+@cuda.jit
+def gpu_round(in_col, out_col, decimal):
+    i = cuda.grid(1)
+    round_val = 10 ** (-1.0 * decimal)
+
+    if i < in_col.size:
+        newval = in_col[i] // round_val * round_val
+        remainder = fmod(in_col[i], round_val)
+
+        if remainder > (.5 * round_val) and in_col[i] > 0:
+            newval = newval + round_val
+            out_col[i] = newval
+
+        elif abs(remainder) < (.5 * round_val) and in_col[i] < 0:
+            newval = newval + round_val
+            out_col[i] = newval
```

Comment:
Looks like both the `if` and `elif` blocks are doing the same thing

Reply:
I should make this logic more clear. There's undoubtedly an easier way to enumerate it.

If the original value is negative, floor division will round "down" differently than if it's positive.

I want to add the round_val if
- The original value was positive and the last digit is > 5
- The original value is negative and the last digit is < 5

Checking the value of the remainder is a way to verify whether the last digit is < or > 5. But, the remainder will be negative for negative original values, so using `abs` is one way to make this easier. 

To show an example:

```
x = -1.534
y = 1.234
new_x = x // .01 * .01
new_y = y // .01 * .01
print(new_x, new_y)
modx = math.fmod(x, .01)
mody = math.fmod(y, .01)
print(modx, mody)

print('real answer:', np.round(x, 2), np.round(y, 2))
-1.54 1.23
-0.003999999999999998 0.00399999999999996
real answer: -1.53 1.23
```

modx is -0.004, and the original x is negative, so I want to round this "down" to -1.53 not -1.54, putting me in the second branching logic.

Let me know if this could be more clear

Reply:
I think we could just set a `multiplier` of 1 or -1 depending on if the input is positive or negative. Do the rounding against the absolute value and then multiply by the `multiplier`, but I'm not sure how much simpler than would make this logic.